app/src/main/java/io/neurolab/MainActivity.java

```diff
@@ -50,39 +58,60 @@ protected void onCreate(Bundle savedInstanceState) {
 
         NavigationView navigationView = findViewById(R.id.nav_view);
         navigationView.setNavigationItemSelectedListener(this);
-    }
 
-    public void moveRocket(View view){
-        if(!moving){
-            float PivotX = rocketimage.getPivotX();
-            float PivotY = rocketimage.getPivotY();
-
-            Animation launch = new TranslateAnimation(0,0, lastPos, newPos);
-            launch.setDuration(1000);
-            launch.setFillAfter(true);
-            launch.setRepeatCount(10);
-            launch.setAnimationListener(new Animation.AnimationListener() {
-                @Override
-                public void onAnimationStart(Animation animation) {
-                    moving = true;
-                }
-
-                @Override
-                public void onAnimationEnd(Animation animation) {
-                    moving = false;
-                }
-
-                @Override
-                public void onAnimationRepeat(Animation animation) {
-
-                }
-            });
-            int c = lastPos;
-            lastPos = newPos;
-            newPos = c;
-
-            rocketimage.startAnimation(launch);
-        }
+        focusButton = findViewById(R.id.btn_focus);
+        relaxButton = findViewById(R.id.btn_relax);
+        vjButton = findViewById(R.id.btn_vj);
+        serialButton = findViewById(R.id.btn_serial);
+
+        // Setting Listeners of the various program buttons
+        focusButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(MainActivity.this, R.string.focus_toast, Toast.LENGTH_SHORT).show();
+                Intent intent = new Intent(MainActivity.this, ProgramModeActivity.class);
+                Bundle bundle = new Bundle();
+                bundle.putInt(INTENT_KEY_PROGRAM_MODE, FOCUS_PROGRAM_MODE);
+                intent.putExtras(bundle);
+                startActivity(intent);
+            }
+        });
+
+        relaxButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(MainActivity.this, R.string.relax_toast, Toast.LENGTH_SHORT).show();
+                Intent intent = new Intent(MainActivity.this, ProgramModeActivity.class);
+                Bundle bundle = new Bundle();
+                bundle.putInt(INTENT_KEY_PROGRAM_MODE, RELAX_PROGRAM_MODE);
+                intent.putExtras(bundle);
+                startActivity(intent);
```

Comment:
Create a single function with 2 params - R.string.relax_toast and RELAX_PROGRAM_MODE

Reuse for 4 listeners